cli/cook/subcommands/show.py

```diff
@@ -0,0 +1,338 @@
+
+    def satisfy_pred():
+        resp_ = make_request_fn(cluster, uuids)
+        return pred(resp_.json())
+
+    try:
+        resp = make_request_fn(cluster, uuids)
+        if resp.status_code == requests.codes.ok:
+            entities = resp.json()
+            if pred and not pred(entities):
+                entities = wait_until(satisfy_pred, timeout, interval)
+                if not entities:
+                    raise Exception('Timeout waiting for response')
+            return entities
+        else:
+            return {}
+    except requests.exceptions.ConnectionError as ce:
+        logging.info(ce)
+        return {}
+
+
+def make_entity_request(cluster, endpoint, params):
+    """Attempts to query entities corresponding to the given params from cluster."""
+    url = make_url(cluster, endpoint)
+    resp = http.get(url, params=params)
+    logging.info('response from cook: %s' % resp.text)
+    return resp
+
+
+def make_job_request(cluster, uuids):
+    """Attempts to query jobs corresponding to the given uuids from cluster."""
+    return make_entity_request(cluster, 'rawscheduler', params={'job': uuids, 'partial': 'true'})
+
+
+def make_instance_request(cluster, uuids):
+    """Attempts to query instances corresponding to the given uuids from cluster."""
+    return make_entity_request(cluster, 'rawscheduler', params={'instance': uuids, 'partial': 'true'})
+
+
+def make_group_request(cluster, uuids):
+    """Attempts to query groups corresponding to the given uuids from cluster."""
+    return make_entity_request(cluster, 'group', params={'uuid': uuids, 'partial': 'true', 'detailed': 'true'})
+
+
+def query_entities(cluster, uuids, pred_jobs, pred_instances, pred_groups, timeout, interval,
+                   include_jobs=True, include_instances=True, include_groups=True):
+    """Queries cluster for the given uuids, searching (by default) for jobs, instances, and groups."""
+    count = 0
+    entities = {}
+    if include_jobs:
+        entities['jobs'] = query_cluster(cluster, uuids, pred_jobs, timeout, interval, make_job_request)
+        count += len(entities['jobs'])
+    if include_instances:
+        entities['instances'] = query_cluster(cluster, uuids, pred_instances, timeout, interval, make_instance_request)
+        count += len(entities['instances'])
+    if include_groups:
+        entities['groups'] = query_cluster(cluster, uuids, pred_groups, timeout, interval, make_group_request)
+        count += len(entities['groups'])
+    entities['count'] = count
+    return entities
+
+
+def query(clusters, uuids, pred_jobs=None, pred_instances=None, pred_groups=None, timeout=None, interval=None):
+    """
+    Attempts to query entities from the given clusters. The uuids are provided in args. Optionally
+    accepts a predicate, pred, which must be satisfied within the timeout.
+    """
+    count = 0
+    all_entities = {'clusters': {}}
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
Should we default `max_workers` to processor count?

Reply:
Changed this to the number of usable CPUs, `len(os.sched_getaffinity(0))`